api/storage_service.cc

```diff
@@ -1540,9 +1540,16 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
 
     ss::upgrade_to_raft_topology.set(r,
             [&ss] (std::unique_ptr<http::request> req) -> future<json::json_return_type> {
-        co_await ss.invoke_on(0, [] (auto& ss) {
-            return ss.start_upgrade_to_raft_topology();
-        });
+        apilog.info("Upgrade to raft topology");
+        try {
+            co_await ss.invoke_on(0, [] (auto& ss) {
+                return ss.start_upgrade_to_raft_topology();
+            });
+        } catch (...) {
+            auto ex = std::current_exception();
+            apilog.error("Upgrade to raft topology failed: {}", ex);
```

Comment:
This endpoint, despite its name in the code, does not block until the upgrade is complete, it only tells the cluster to start upgrade to raft topology. The message that `start_upgrade_to_raft_topology` prints before finishing successfully is

```c++
    rtlogger.info("upgrade to topology on raft is scheduled");
```

So I suggest to rephrase this message (and the one on line 1543) not to imply that this endpoint does the full upgrade, I think it is confusing right now. So what about `"Requested to schedule upgrade to raft topology"` / `"Failed to schedule upgrade to raft topology: {}"`?

Reply:
Sounds good.  I'll apply and send v2